dts/riscv/hpmicro/hpm6750.dtsi

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/clock/hpmicro-clocks.h>
+#include <mem.h>
+
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cpu0: cpu@0 {
+			compatible = "riscv";
+			device_type = "cpu";
+			reg = <0>;
+			status = "okay";
+			riscv,isa = "rv32imafdcxandes";
+			i-cache-line-size = <32>;
+			d-cache-line-size = <32>;
+			CPU0_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+		cpu1: cpu@1 {
+			compatible = "riscv";
+			device_type = "cpu";
+			reg = <1>;
+			status = "okay";
+			riscv,isa = "rv32imafdcxandes";
+			i-cache-line-size = <32>;
+			d-cache-line-size = <32>;
+			CPU1_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+
+		fgpio: peripheral@C0000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xC0000 0x2000>;
+		};
+	};
+
+	soc {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "hpmicro,hpm6750";
+		ranges;
+		clk: clock-controller@f4100000 {
+			compatible = "hpmicro,hpm-clock";
+			reg = <0xf4100000 0x100>;
+			clock-frequency = <816000000>;
+			reg-names = "clk";
+			status = "okay";
+			clock-sys-core = <HPM_CLOCK_CPU0>;
+			ram-up-time = <(32*1000*9)>;
+			sysctl-present = <2>;
+			#clock-cells = <3>;
+		};
+
+		mtimer: timer@e6000000 {
+			compatible = "andestech,machine-timer";
+			reg = <0xe6000000 0x10>;
+			interrupts-extended = <&CPU0_intc 7 &CPU1_intc 7>;
+		};
+
+		flash0: flash@80000000 {
+			compatible = "soc-nv-flash";
+			reg = <0x80000000 DT_SIZE_M(16)>;
+		};
+
+		ilm: memory@0 {
+			compatible = "zephyr,memory-region";
+			reg = <0x0 DT_SIZE_K(256)>;
+			zephyr,memory-region = "ITCM";
+		};
+
+		dlm: memory@80000 {
+			compatible = "zephyr,memory-region";
+			reg = <0x080000 DT_SIZE_K(256)>;
+			zephyr,memory-region = "DTCM";
+		};
+
+		sram: memory@1080000 {
+			device_type = "memory";
+			compatible = "mmio-sram";
+			reg = <0x01080000 DT_SIZE_K(1024)>;
+		};
+
+		core0_lm_slv: memory@1000000 {
+			device_type = "memory";
+			reg = <0x1000000 DT_SIZE_K(512)>;
+		};
+
+		core1_lm_slv: memory@1180000 {
+			device_type = "memory";
+			reg = <0x1180000 DT_SIZE_K(256)>;
+		};
+
+		noncache: memory@11c0000 {
```

Comment:
I don't think it's necessary to establish a noncache node.
Using zephyr/arch/common/nocache.ld instead, it will automatic distribution on demand. I have tested by using nocache.ld in ethernet driver.